esp/logging/datafieldmap.cpp

```diff
@@ -0,0 +1,101 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "datafieldmap.hpp"
+const char* const defaultLogSourcePath = "Source";
+
+void ensureInputString(const char* input, bool lowerCase, StringBuffer& inputStr, int code, const char* msg)
```

Comment:
since the 3rd parameter is actually an outpu, it might make more sense to rename